File: ERP.Domain/Entities/SocioNegocio.cs
namespace ERP.Domain.Entities;

public class SocioNegocio
{
    public int Id { get; set; }
    public string TipoSocio { get; set; } = string.Empty;
    public string TipoDocumento { get; set; } = string.Empty;
    public string NumeroDocumento { get; set; } = string.Empty;
    public string Nombres { get; set; } = string.Empty;
    public string? Direccion { get; set; }
    public string? Telefono { get; set; }
    public string? Email { get; set; }
    public decimal LimiteCredito { get; set; }
    public decimal Saldo { get; set; }
    public bool Activo { get; set; } = true;
    public DateTime FechaRegistro { get; set; } = DateTime.UtcNow;
}

File: ERP.Frontend/src/types/socioNegocio.ts
export interface SocioNegocio {
  id: number; // 🔹 obligatorio
  tipoSocio: "C" | "P";
  tipoDocumento: string;
  numeroDocumento: string;
  nombres: string;
  direccion?: string;
  telefono?: string;
  email?: string;
  limiteCredito: number;
  saldo: number;
  activo: boolean;
  fechaRegistro?: string;
}

export interface SocioNegocioForm extends Partial<SocioNegocio> {
  id?: number; // opcional en el formulario
}
